### PR TITLE
Fixes missing curly braces in statements.

### PR DESCRIPTION
**Pull request by EstherDarkish**
Added a new esLint rule to force curly braces in statements.

To fix and update the code, I used Intellij's `Code => Analyse Code => Run Inspection by Name => "Statement body without braces"` on Gridstudy's *.js files to find these violations.

### CODE REVIEW

File: src/components/security-analysis-result.js [hunk 1]
```diff
@@ -180,9 +180,17 @@ const SecurityAnalysisResult = ({ onClickNmKConstraint, result }) => {
         function sortAndAddResults(result, array) {
             const compareValue = (a, b) => {
                 const mult = reverse ? 1 : -1;
-                if (a === undefined && b === undefined) return 0;
-                else if (b === undefined) return -mult;
-                else if (a === undefined) return mult;
+                if (a === undefined && b === undefined) {
+                    return 0;
+                } else {
```
Comment (laurgarn) [inline src/components/security-analysis-result.js:185]: Here, in addition to using more vertical space, it uses more horizontal space :grin:
-> `else if` is not forbidden

Comment (EstherDarkish) [reply src/components/security-analysis-result.js:185]: Space is not an issue. maintainability is.

File: src/components/dialogs/inputs/input-hooks.js [hunk 2]
```diff
@@ -706,18 +710,26 @@ export const useValidNodeName = ({ studyUuid, defaultValue, triggerReset }) => {
...
+            setAdornment(null);
+        }
+        if (checking) {
             setAdornment(inputAdornment(<CircularProgress size="1rem" />));
-        else if (!isValidName) setAdornment(undefined);
-        else
-            setAdornment(
-                inputAdornment(<CheckIcon style={{ color: 'green' }} />)
-            );
+        } else {
```
Comment (laurgarn) [inline src/components/dialogs/inputs/input-hooks.js:718]: `if else` stair

File: src/components/dialogs/inputs/reactive-capability-curve-table.js [hunk 3]
```diff
@@ -294,17 +294,22 @@ export const useReactiveCapabilityCurveTableValues = ({
...
-                    // the fields are required if new intermediate lines are added
-                    if (
-                        displayedPreviousValues === undefined ||
-                        displayedPreviousValues[index]?.p === ''
-                    )
-                        return true;
+                else {
+                    if (index === values.length - 1 || index === 0) {
+                        return undefined;
+                    } else {
```
Comment (laurgarn) [inline src/components/dialogs/inputs/reactive-capability-curve-table.js:304]: `if else` stair

File: src/components/dialogs/inputs/reactive-capability-curve-table.js [hunk 4]
```diff
@@ -329,10 +334,15 @@ export const useReactiveCapabilityCurveTableValues = ({
                 {displayedValues.map((value, index, displayedValues) => {
                     const id = getId(value);
                     let labelSuffix;
-                    if (index === 0) labelSuffix = 'min';
-                    else if (index === displayedValues.length - 1)
-                        labelSuffix = 'max';
-                    else labelSuffix = index.toString();
+                    if (index === 0) {
+                        labelSuffix = 'min';
+                    } else {
```
Comment (laurgarn) [inline src/components/dialogs/inputs/reactive-capability-curve-table.js:339]: `if else` stair

File: src/components/refactor/dialogs/generator/reactive-limits/reactive-capability-curve/reactive-capability-curve-table.js [hunk 5]
```diff
@@ -59,10 +61,15 @@ export const ReactiveCapabilityCurveTable = ({
 
             {rows.map((value, index, displayedValues) => {
                 let labelSuffix;
-                if (index === 0) labelSuffix = 'min';
-                else if (index === displayedValues.length - 1)
-                    labelSuffix = 'max';
-                else labelSuffix = index.toString();
+                if (index === 0) {
+                    labelSuffix = 'min';
+                } else {
```
Comment (laurgarn) [inline src/components/refactor/dialogs/generator/reactive-limits/reactive-capability-curve/reactive-capability-curve-table.js:66]: `if else` stair

File: src/components/refactor/dialogs/generator/set-points/set-points-form.js [hunk 6]
```diff
@@ -46,11 +46,15 @@ const SetPointsForm = ({
             previousValues?.voltageRegulatorOn === true);
 
     const previousRegulation = () => {
-        if (previousValues?.voltageRegulatorOn)
+        if (previousValues?.voltageRegulatorOn) {
             return intl.formatMessage({ id: 'On' });
-        else if (previousValues?.voltageRegulatorOn === false)
-            return intl.formatMessage({ id: 'Off' });
-        else return null;
+        } else {
```
Comment (laurgarn) [inline src/components/refactor/dialogs/generator/set-points/set-points-form.js:51]: `if else` stair

File: src/components/util/text-field-with-adornment.js [hunk 7]
```diff
@@ -42,9 +42,15 @@ const TextFieldWithAdornment = (props) => {
 
     const getAdornmentClassName = useCallback(
         (variant) => {
-            if (variant === 'filled') return classes.adornRightFilled;
-            else if (variant === 'standard') return classes.adornRightOther;
-            else return null;
+            if (variant === 'filled') {
+                return classes.adornRightFilled;
+            } else {
```
Comment (laurgarn) [inline src/components/util/text-field-with-adornment.js:47]: `if else` stair

File: src/components/dialogs/inputs/boolean.js [hunk 8]
```diff
@@ -94,8 +94,13 @@ export const useNullableBooleanValue = ({
     const handleChangeValue = useCallback(
         (event) => {
             setValue((oldVal) => {
-                if (oldVal) return null;
-                else if (oldVal === null) return false;
+                if (oldVal) {
+                    return null;
+                } else {
```
[the remark this answers is not in the document]
Comment (etiennehomer) [reply src/components/dialogs/inputs/boolean.js:99]: What do you mean ? It could be inline ?

[the remark this answers is not in the document]
Comment (laurgarn) [reply src/components/dialogs/inputs/boolean.js:99]: `} else { if (...) {...} }` could be simplified keeping the `else if` with `} else if (...) {...}`

[the remark this answers is not in the document]
Comment (EstherDarkish) [reply src/components/dialogs/inputs/boolean.js]: Fixed. Intellij was overzealous.

File: src/components/dialogs/inputs/input-hooks.js (hunk 2 above)
Comment (EstherDarkish) [reply src/components/dialogs/inputs/input-hooks.js:720]: Fixed.

File: src/components/dialogs/inputs/reactive-capability-curve-table.js (hunk 4 above)
Comment (EstherDarkish) [reply src/components/dialogs/inputs/reactive-capability-curve-table.js:338]: Fixed.

File: src/components/dialogs/inputs/reactive-capability-curve-table.js (hunk 3 above)
Comment (EstherDarkish) [reply src/components/dialogs/inputs/reactive-capability-curve-table.js]: Fixed.

File: src/components/refactor/dialogs/generator/reactive-limits/reactive-capability-curve/reactive-capability-curve-table.js (hunk 5 above)
Comment (EstherDarkish) [reply src/components/refactor/dialogs/generator/reactive-limits/reactive-capability-curve/reactive-capability-curve-table.js:68]: Fixed.

File: src/components/refactor/dialogs/generator/set-points/set-points-form.js (hunk 6 above)
Comment (EstherDarkish) [reply src/components/refactor/dialogs/generator/set-points/set-points-form.js]: Fixed.

**Review by laurgarn — APPROVED** [review]

**Pull request merged**